### ISSUE TITLE
[Monitoring V1] [Istio V1] hide the old version in Rancher v2.5.4

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#30507]
We need to add `rancher_max_verson: 2.5.3` to 
- Monitoring v1 chart `0.1.4` 
- Istio v1 chart `1.4.1000` 

to stop showing them on rancher v2.5.4 setup.

**Comment by maggieliu** [comment]
Why do we need to hide 0.1.4? I think we need 0.1.4 because for users who updated from 2.4 with k8s 1.15 they need to see the 0.1.4 chart for monitoring. @jiaqiluo

**Comment by maggieliu** [comment]
We do want to show 0.1.4 because this is the only version of monitoring v1 that support k8s 1.15 for users upgrading from 2.4 with k8s 1.15.

**Comment by maggieliu** [comment]
@cbron Do you think we need to hide the older Istio version?

**Comment by cbron** [comment]
@jiaqiluo was there a specific reason to deprecate Istio 1.4.1000 ? I don't see how the monitoring issue affects it. 

I'm not sure it really matters, it might be good to deprecate it in general so people stop using the older 1.4 line and use either the (deprecated and unsupported-upstream) 1.5 line or the v2 line. We can't max 1.5 or it would stop the v1 line altogether. 

@brendarearden can you add a max-version to istio 1.4.1000 of 2.5.3.

**Comment by jiaqiluo** [comment]
@maggieliu @cbron 
In general, for the v1 charts, we show only 1 version in the UI. 

This might be a better way to achieve it:
- For monitoring 0.1.4, instead of removing it on v2.5.4, we can add the `kubeVesion` parameter to its chart to make it only available for <=1.15 cluster. As the result, the 1.15 cluster can see both 0.14 and 0.2.0 in the UI, the 1.16+ cluster can see only 0.2.0 
- For Istio, the only reason is to make sure we show only 1 version. 

How do you think?

**Comment by maggieliu** [comment]
@jiaqiluo I am ok with using the kubeVersion parameter on the 0.1.4 chart. @aiyengar2 can you see if that's possible to update the 0.1.4 to only show for 1.15 k8s (will this break 2.4?)? Or do we need a new chart?

**Comment by aiyengar2** [comment]
> @aiyengar2 can you see if that's possible to update the 0.1.4 to only show for 1.15 k8s (will this break 2.4?)? Or do we need a new chart? 

I had to work out the different scenarios to figure out the issue. Here are the upgrade paths:

#### Scenarios

Initial Circumstance:
1. User has a k8s 1.16+ cluster in Rancher 2.4.x before kubeVersions are respected.
2. User has a k8s 1.15+ cluster in Rancher 2.4.x before kubeVersions are respected.

Monitoring Upgrade (per circumstance):
- No upgrade. Tries to update to Monitoring v0.1.4
  - Upgrades to Rancher 2.4.x after kubeVersions are respected. Tries to enable / disable Monitoring
    - Upgrades to Rancher 2.5.x before kubeVersions are respected. Tries to enable / disable Monitoring
      - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
    - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
  - Upgrades to Rancher 2.5.x before kubeVersions are respected. Tries to enable / disable Monitoring
    - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
  - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
- Upgrades Rancher 2.4.x after kubeVersions are respected. Tries to update to Monitoring v0.1.4
  - Upgrades Rancher 2.5.x before kubeVersions are respected. Tries to enable / disable Monitoring
    - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
  - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
- Upgrades Rancher 2.5.x before kubeVersions are respected. Tries to update to Monitoring v0.1.4 
  - Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
- Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to update to Monitoring v0.1.4

#### Proposed Change: kubeVersion

We update the kubeVersion (a [semver range](https://github.com/rancher/system-charts/blob/6e7a4048a4c71e0c38a0f3c9a08d177d9ca0733d/charts/rancher-monitoring/v0.2.0/Chart.yaml#L12)) to `<1.16.0-0` on the Rancher v0.1.4 chart, both in `dev-v2.4` and `dev-v2.5`.

Since kubeVersion won't ever prevent k8s 1.15 clusters from installing v0.1.4, all scenarios work.

For k8s 1.16 clusters, these are the paths that would break (e.g. anything that goes from not respecting kubeVersions to respecting kubeVersions)
- No upgrade. Tries to update to Monitoring v0.1.4
  - [BREAK] Upgrades to Rancher 2.4.x after kubeVersions are respected. Tries to enable / disable Monitoring
  - Upgrades to Rancher 2.5.x before kubeVersions are respected. Tries to enable / disable Monitoring
    - [BREAK] Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
  - [BREAK] Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring
- Upgrades Rancher 2.5.x before kubeVersions are respected. Tries to update to Monitoring v0.1.4 
  - [BREAK] Upgrades to Rancher 2.5.x after kubeVersions are respected. Tries to enable / disable Monitoring

#### Additional Change: rancher_min_version

To whackamole the upgrade situations with k8s 1.16, we can introduce the following additional change:
- [`rancher_min_version` in `dev-v2.4`](https://github.com/rancher/system-charts/blob/13e80a317f6affb07c3a3262c5ff71c7d49eef34/charts/rancher-monitoring/v0.1.4/questions.yml#L1) to `2.4.13-rc1` (blocks situation 1 that is broken above)
- [`rancher_min_version` in `dev-v2.5`](https://github.com/rancher/system-charts/blob/244f557b90629b19d3642af04375ed4b78b7ae8e/charts/rancher-monitoring/v0.1.4/questions.yml#L1) to `2.5.4-rc1` (blocks situation 2 that is broken above)

However, that has the following unintended consequences to k8s 1.15 clusters (since they could upgrade from 2.4.x to 2.5.0-2.5.3):
- Upgrades Rancher 2.4.x after kubeVersions are respected. Tries to update to Monitoring v0.1.4
  - [BREAK] Upgrades Rancher 2.5.x before kubeVersions are respected. Tries to enable / disable Monitoring

#### TLDR

Can we enforce that a user who migrates from Rancher 2.4.x to Rancher 2.5 with a k8s 1.16 cluster using v0.1.4 must upgrade their monitoring version to v0.2.0 before the migration? I'm not sure if this is enforceable, so I don't know if we can prevent upgrade issues from popping up.

**Comment by maggieliu** [comment]
@jiaqiluo It sounds like we need 0.1.4 monitoring v1 in 2.5.4 to support upgrade scenario from 2.4 and hiding it with the kubeVersion doesn't resolve the issue. Is there an issue with showing both?

**Comment by jiaqiluo** [comment]
@aiyengar2 @maggieliu 

I am ok with showing 2 monitoring charts for all k8s versions on rancher:`v2.5.4`

As the consultation, on rancher:`v2.5.4`:
- both monitoring 0.1.4 and 0.2.0 will be available
- only the new Istio chart 1.5.900 is available

**Comment by jiaqiluo** [comment]
wait for a fix for the [A typo in the line](https://github.com/rancher/system-charts/blob/244f557b90629b19d3642af04375ed4b78b7ae8e/charts/rancher-istio/1.4.1000/questions.yaml#L4) to make it work.

**Comment by jiaqiluo** [comment]
This is validated on rancher:v2.5.4-rc6 

Steps:
- run rancher single install
- add a cluster with k8s v1.16.15


results:
- for monitoring v1, 2 versions show in the list 
![image](https://user-images.githubusercontent.com/6218999/102519436-b77dc880-404f-11eb-9777-1456abcf51e7.png)

- for istio v1, 1 version shows in the list 
![image](https://user-images.githubusercontent.com/6218999/102519499-c82e3e80-404f-11eb-9b02-9e904182453c.png)
